### DISCUSSION TITLE
Activity is logged when the user logs out but not when the user logs in.

### DISCUSSION THREAD

**Discussion by alantucker** [spatie/laravel-activitylog#952] [Q&A] [answered]
Hi

I am using v3 and logging the user model. I have noticed that activity is logged when the user logs out but not when the user logs in.

When checking the database, it has a description entry of 'updated', and the properties entry has both 'old' and 'attributes' data the same. In this case, I am tacking name and email.

Q) Does anyone know why this is just on log out and not login?? Is this to do with the sessions?

Thanks for your time.

**Comment by Gummibeer** [comment, marked as answer]
Hey,

depending on your used auth logic the `remember_me` Token is updated during sign out for example. Depending on your [Model Activity logging configuration](https://spatie.be/docs/laravel-activitylog/v4/advanced-usage/logging-model-events#basics-of-logging-configuration) there will be logs with empty changes as any field updated but isn't in the list of attributes to be logged and you haven't disabled empty logs.

So you should check your auth logic where an update of the user model happens and also your model logging configuration to see what and when is logged.

In case you want explicit sign in/out activities you should use manual logging using the `activity()` global function.

**Reply by alantucker** [reply]
Thanks for the reply.

I am kinda new to using Laravel. For auth I am using Jetstream and as far as the model goes I have the following:

    protected static $logAttributes = ['name', 'email'];

    protected static $logName = 'user';

    public function getDescriptionForEvent(string $eventName): string
    {
        return "The user has been {$eventName}";
    }

Thanks again
